web/packages/teleport/src/Discover/Kubernetes/SetupAccess/SetupAccess.story.tsx

```diff
@@ -53,6 +59,27 @@ export const SsoUser = () => (
   </MemoryRouter>
 );
 
+function setAutoDiscovery(props: State): State {
```

Comment:
i'm not sure this is necessary, think you can just do this?

```
export const WithTraitsAutoDiscovery = () => (
  <MemoryRouter>
    <SetupAccess {...props}  agentMeta={...props.agentMeta, autoDiscovery= .....} />
  </MemoryRouter>
);
```

i wasn't sure if you knew, but this copy is a shallow copy, nested things dont get copied and you could be modifying the original

Reply:
Yep, it's shallow copy, that's why we did it twice so we don't change original `props`. I didn't figure out syntax at first for the inline change (it needs extra braces `agentMeta={{...props.agentMeta, autoDiscovery: {...} }}` ), changed. https://github.com/gravitational/teleport/pull/38432/commits/0fd4f00a54491979719d7983241dacbd3aa28518